solutions/src/main/java/ru/sberbank/school/task10/ConcurrentFixedPool.java

```
@@ -0,0 +1,121 @@
+        this.tasks = new ArrayBlockingQueue<>(scale * 10);
+        this.stopped = false;
+    }
+
+    @Override
+    public void start() {
+        lock.lock();
+        try {
+            if (!threads.isEmpty() && !stopped) {
+                throw new IllegalStateException("This method is already start!");
+            }
+
+            for (int i = 0; i < scale; i++) {
+                newThread(i).start();
+            }
+        } finally {
+            lock.unlock();
+            stopped = false;
+        }
+
+    }
+
+    @Override
+    public void stopNow() {
+
+        if (stopped) {
+            throw new IllegalStateException("This method is already invoke!");
+        }
+
+        lock.lock();
+        try {
+            tasks.clear();
+
+            for (Thread t : threads) {
+                t.interrupt();
+            }
+
+            threads.clear();
+        } finally {
+            lock.unlock();
+            stopped = true;
+        }
+
+
+    }
+
+    @Override
+    public void execute(Runnable runnable) {
+
+        if (threads.isEmpty() && stopped) {
+            throw new NoSuchElementException("No threads running! Call start() first");
+        }
+
+        tasks.add(runnable);
+        tasks.notify();
+
+    }
+
+    private Thread newThread(int i) {
+        Thread thread = new Thread("Thread-" + (i + 1)) {
+
+            @Override
+            public void run() {
+                Runnable runnable;
+                while (!Thread.interrupted()) {
+
+                    if (tasks.isEmpty()) {
+                        try {
+                            tasks.wait();
+                        } catch (InterruptedException e) {
+                            Thread.currentThread().interrupt();
+                        }
+                    }
+                    runnable = tasks.poll();
+
+                    try {
+                        runnable.run();
+                    } catch (NullPointerException ignored) {
```

Empty catch block.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a81415117>*